rust/Makefile

```diff
@@ -60,11 +61,73 @@ alloc-cfgs = \
     --cfg no_sync \
     --cfg no_thin
 
+proc_macro2-skip_flags := \
+    --edition=2021 \
+    -Drust_2018_idioms \
+    -Dunreachable_pub \
+    -Dunsafe_op_in_unsafe_fn
+
+proc_macro2-flags := \
+    --edition=2018 \
+    -Amissing_docs \
+    --cfg 'feature="proc-macro"' \
+    --cfg use_proc_macro \
+    --cfg wrap_proc_macro
+
+quote-skip_flags := \
+    --edition=2021 \
+    -Drust_2018_idioms
+
+quote-flags := \
+    --edition=2018 \
+    -Amissing_docs \
+    --extern proc_macro2 \
+    --cfg 'feature="proc-macro"'
+
+syn-skip_flags := \
+    --edition=2021 \
+    -Drust_2018_idioms \
+    -Dunreachable_pub \
+    -Dunsafe_op_in_unsafe_fn
+
+syn-flags := \
+    --edition=2018 \
+    -Amissing_docs \
+    --cfg 'feature="clone-impls"' \
+    --cfg 'feature="derive"' \
+    --cfg 'feature="extra-traits"' \
+    --cfg 'feature="fold"' \
+    --cfg 'feature="full"' \
+    --cfg 'feature="parsing"' \
+    --cfg 'feature="printing"' \
+    --cfg 'feature="proc-macro"' \
+    --cfg 'feature="quote"' \
+    --cfg 'feature="visit"' \
+    --cfg 'feature="visit-mut"'
+
+serde_derive-skip_flags := \
+    --edition=2021 \
+    -Drust_2018_idioms \
+    -Dunreachable_pub
+
+serde_derive-flags := \
+    -Amissing_docs
+
+serde-skip_flags := \
+    --edition=2021 \
+    -Drust_2018_idioms \
+    -Dunreachable_pub
+
+serde-flags := \
+    -Amissing_docs \
+    --cfg no_fp_fmt_parse
```

Comment:
If you pass `--cfg 'feature="derive"' --extern serde_derive`, you don't need `--extern serde_derive` elsewhere, but can instead access the derive macros using `use serde::{Serialize, Deserialize};`.